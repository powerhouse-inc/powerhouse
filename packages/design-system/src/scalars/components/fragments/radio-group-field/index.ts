export * from "./radio-group.js";
export * from "./radio-group-field.js";
