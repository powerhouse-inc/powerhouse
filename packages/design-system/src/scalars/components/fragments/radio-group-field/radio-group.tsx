import React, { useId } from "react";
import * as RadioGroupPrimitive from "@radix-ui/react-radio-group";

export interface CustomizableRadioGroupProps
  extends React.ComponentPropsWithoutRef<typeof RadioGroupPrimitive.Root> {
  children: React.ReactNode;
  className?: string;
  id?: string;
  name?: string;
}

export const RadioGroup = React.forwardRef<
  React.ElementRef<typeof RadioGroupPrimitive.Root>,
  CustomizableRadioGroupProps
>(({ children, className, id: propId, name, ...props }, ref) => {
  const prefix = useId();
  const id = propId ?? `${prefix}-radio-group`;

  return (
    <RadioGroupPrimitive.Root
      className={className}
      id={id}
      name={name}
      {...props}
      ref={ref}
    >
      {children}
    </RadioGroupPrimitive.Root>
  );
});

RadioGroup.displayName = "RadioGroup";
