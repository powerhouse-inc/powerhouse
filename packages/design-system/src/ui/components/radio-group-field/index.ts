export * from "./radio-group-field.js";
export * from "./radio-group.js";
export * from "./radio.js";
