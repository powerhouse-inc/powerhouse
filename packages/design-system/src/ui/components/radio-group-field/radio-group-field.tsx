import type {
  FieldErrorHandling,
  InputBaseProps,
  RadioGroupProps,
} from "@powerhousedao/design-system";
import {
  cn,
  FormLabel,
  FormMessageList,
  Radio,
  RadioGroup,
  withFieldValidation,
} from "@powerhousedao/design-system";
import { type HTMLAttributes, type Ref, forwardRef, useId } from "react";
type RadioGroupFieldBaseProps = Omit<
  HTMLAttributes<HTMLDivElement>,
  | keyof InputBaseProps<string>
  | keyof FieldErrorHandling
  | keyof RadioGroupProps
  | "dir"
> & {
  dir?: "ltr" | "rtl";
};

export type RadioGroupFieldProps = RadioGroupFieldBaseProps &
  InputBaseProps<string> &
  FieldErrorHandling &
  RadioGroupProps;

const RadioGroupFieldRaw = forwardRef<HTMLDivElement, RadioGroupFieldProps>(
  (
    {
      autoFocus,
      className,
      defaultValue,
      description,
      disabled,
      warnings = [],
      errors = [],
      id: propId,
      label,
      name,
      onChange,
      options = [],
      required,
      value,
      ...props
    },
    ref,
  ) => {
    const hasLabel = label !== undefined;
    const hasError = errors.length > 0;
    const prefix = useId();
    const id = propId ?? `${prefix}-radio-group`;

    return (
      <RadioGroup
        aria-invalid={hasError}
        aria-label={!hasLabel ? "Radio group" : undefined}
        aria-required={required}
        autoFocus={autoFocus}
        className={cn("flex flex-col gap-2", className)}
        defaultValue={defaultValue}
        id={id}
        name={name}
        onValueChange={(newValue) => {
          onChange?.(newValue);
        }}
        value={value}
        disabled={disabled}
        {...props}
        ref={ref}
      >
        {hasLabel && (
          <FormLabel
            description={description}
            hasError={hasError}
            htmlFor={id}
            required={required}
            disabled={disabled}
          >
            {label}
          </FormLabel>
        )}
        {options.map((option, index) => (
          <div
            key={`${prefix}-radio-${index}-${option.value}`}
            className="flex items-center gap-2"
            role="presentation"
          >
            <Radio
              id={`${prefix}-radio-${index}-${option.value}`}
              label={option.label}
              value={option.value}
              description={option.description}
              disabled={disabled || option.disabled}
              hasError={hasError}
            />
          </div>
        ))}
        {warnings.length > 0 && (
          <FormMessageList messages={warnings} type="warning" />
        )}
        {hasError && <FormMessageList messages={errors} type="error" />}
      </RadioGroup>
    );
  },
);

export const RadioGroupField = forwardRef(function RadioGroupField(
  props: RadioGroupFieldProps,
  ref: Ref<HTMLDivElement>,
) {
  const Component =
    withFieldValidation<RadioGroupFieldProps>(RadioGroupFieldRaw);
  return <Component {...props} ref={ref} />;
});

RadioGroupField.displayName = "RadioGroupField";
