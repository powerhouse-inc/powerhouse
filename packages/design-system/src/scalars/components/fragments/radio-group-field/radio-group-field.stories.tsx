import type { Meta, StoryObj } from "@storybook/react";
import { withForm } from "../../../lib/decorators.js";
import {
  getDefaultArgTypes,
  getValidationArgTypes,
  StorybookControlCategory,
} from "../../../lib/storybook-arg-types.js";
import { RadioGroupField } from "./radio-group-field.js";

const meta = {
  title: "Document Engineering/Fragments/RadioGroupField",
  component: RadioGroupField,
  decorators: [withForm],
  parameters: {
    layout: "centered",
    chromatic: {
      disableSnapshot: true,
    },
  },
  tags: ["autodocs"],
  argTypes: {
    ...getDefaultArgTypes(),

    options: {
      control: "object",
      description:
        "Array of options with label, value, description, and disabled state",
      table: {
        type: {
          summary:
            "Array<{ label: string; value: string; description?: string; disabled?: boolean; }>",
        },
        defaultValue: { summary: "[]" },
        category: StorybookControlCategory.COMPONENT_SPECIFIC,
      },
    },

    ...getValidationArgTypes(),
  },
  args: {
    name: "radio-group",
    errors: [],
    warnings: [],
  },
} satisfies Meta<typeof RadioGroupField>;

export default meta;
type Story = StoryObj<typeof meta>;

const defaultOptions = [
  { label: "Option 1", value: "1" },
  { label: "Option 2", value: "2" },
  { label: "Option 3", value: "3" },
];

// Basic examples
export const Default: Story = {
  args: {
    label: "Radio Group",
    options: defaultOptions,
  },
};

export const WithDescription: Story = {
  args: {
    label: "Radio Group",
    description: "This is a helpful description for the Radio Group",
    options: defaultOptions,
  },
};

export const Required: Story = {
  args: {
    label: "Required Radio Group",
    options: defaultOptions,
    required: true,
  },
};

export const WithDefaultValue: Story = {
  args: {
    label: "Radio Group with an option selected by default",
    defaultValue: "2",
    options: defaultOptions,
  },
};

export const Disabled: Story = {
  args: {
    label: "Disabled Radio Group",
    disabled: true,
    options: defaultOptions,
    value: defaultOptions[0].value,
  },
};

// Validation states
export const WithError: Story = {
  args: {
    label: "Radio Group with errors",
    errors: ["Please select an option"],
    options: defaultOptions,
  },
};

export const WithWarning: Story = {
  args: {
    label: "Radio Group with warnings",
    warnings: ["Your selection may need review"],
    options: defaultOptions,
    defaultValue: "3",
  },
};

// Special features
export const WithDescriptionInOptions: Story = {
  args: {
    label: "Radio Group with description in options",
    options: [
      {
        label: "Option 1",
        value: "1",
        description: "Description for option 1",
      },
      {
        label: "Option 2",
        value: "2",
        description: "Description for option 2",
      },
    ],
  },
};

export const WithDisabledOptions: Story = {
  args: {
    label: "Radio Group with disabled options",
    options: [
      { label: "Option 1", value: "1" },
      { label: "Option 2", value: "2", disabled: true },
      { label: "Option 3", value: "3" },
      {
        label: "Option 4",
        value: "4",
        disabled: true,
        description: "This option is disabled",
      },
    ],
  },
};
