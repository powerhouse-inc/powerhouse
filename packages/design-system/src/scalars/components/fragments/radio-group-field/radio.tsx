import React, { useId } from "react";
import * as RadioGroupPrimitive from "@radix-ui/react-radio-group";
import { FormLabel } from "@/scalars/components/fragments/form-label";
import { cn } from "@/scalars/lib/utils";

export interface RadioProps
  extends React.ComponentPropsWithoutRef<typeof RadioGroupPrimitive.Item> {
  className?: string;
  description?: string;
  disabled?: boolean;
  hasError?: boolean;
  id?: string;
  label: string;
  value: string;
}

export const Radio = React.forwardRef<
  React.ElementRef<typeof RadioGroupPrimitive.Item>,
  RadioProps
>(
  (
    {
      className,
      description,
      disabled = false,
      hasError = false,
      id: propId,
      label,
      value,
      ...props
    },
    ref,
  ) => {
    const prefix = useId();
    const id = propId ?? `${prefix}-radio`;

    return (
      <>
        <RadioGroupPrimitive.Item
          aria-disabled={disabled}
          aria-invalid={hasError}
          className={cn(
            "group",
            "aspect-square size-4 rounded-full border-2 border-gray-800",
            "dark:border-gray-400",
            "focus:outline-none focus:ring-2 focus:ring-gray-300/40 focus:ring-offset-0",
            "dark:focus:ring-gray-500/40",
            "hover:border-gray-900",
            "dark:hover:border-gray-50",
            "focus:hover:ring-gray-400/40",
            "dark:focus:hover:ring-gray-400/40",
            disabled && [
              "cursor-not-allowed border-gray-600 opacity-70 hover:border-gray-600",
              "dark:border-gray-600 dark:hover:border-gray-600",
            ],
            hasError && [
              "border-red-700 hover:border-red-900 focus:ring-transparent focus:hover:ring-transparent",
              "dark:border-red-700 dark:hover:border-red-900 dark:focus:ring-transparent dark:focus:hover:ring-transparent",
            ],
            className,
          )}
          disabled={disabled}
          id={id}
          value={value}
          {...props}
          ref={ref}
        >
          <RadioGroupPrimitive.Indicator
            className={cn(
              "relative flex size-full items-center justify-center",
              "after:block after:size-2 after:rounded-full after:bg-gray-800 after:content-['']",
              "dark:after:bg-gray-400",
              "group-hover:after:bg-gray-900",
              "dark:group-hover:after:bg-gray-50",
              disabled && [
                "after:bg-gray-600 after:hover:bg-gray-600",
                "dark:after:bg-gray-600 dark:after:hover:bg-gray-600",
              ],
            )}
          />
        </RadioGroupPrimitive.Item>
        <FormLabel
          description={description}
          disabled={disabled}
          hasError={hasError}
          htmlFor={id}
          required={props.required}
        >
          {label}
        </FormLabel>
      </>
    );
  },
);
