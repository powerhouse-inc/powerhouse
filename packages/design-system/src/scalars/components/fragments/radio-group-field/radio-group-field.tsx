import { cn, FormLabel, FormMessageList, type RadioGroupProps } from "#scalars";
import React, { useId } from "react";
import type { FieldErrorHandling, InputBaseProps } from "../../types.js";
import { withFieldValidation } from "../with-field-validation/with-field-validation.js";
import { RadioGroup } from "./radio-group.js";
import { Radio } from "./radio.js";

type RadioGroupFieldBaseProps = Omit<
  React.HTMLAttributes<HTMLDivElement>,
  | keyof InputBaseProps<string>
  | keyof FieldErrorHandling
  | keyof RadioGroupProps
  | "dir"
> & {
  dir?: "ltr" | "rtl";
};

export type RadioGroupFieldProps = RadioGroupFieldBaseProps &
  InputBaseProps<string> &
  FieldErrorHandling &
  RadioGroupProps;

const RadioGroupFieldRaw = React.forwardRef<
  HTMLDivElement,
  RadioGroupFieldProps
>(
  (
    {
      autoFocus,
      className,
      defaultValue,
      description,
      disabled,
      warnings = [],
      errors = [],
      id: propId,
      label,
      name,
      onChange,
      options = [],
      required,
      value,
      ...props
    },
    ref,
  ) => {
    const hasLabel = label !== undefined;
    const hasError = errors.length > 0;
    const prefix = useId();
    const id = propId ?? `${prefix}-radio-group`;

    return (
      <RadioGroup
        aria-invalid={hasError}
        aria-label={!hasLabel ? "Radio group" : undefined}
        aria-required={required}
        autoFocus={autoFocus}
        className={cn("flex flex-col gap-2", className)}
        defaultValue={defaultValue}
        id={id}
        name={name}
        onValueChange={(newValue) => {
          onChange?.(newValue);
        }}
        value={value}
        disabled={disabled}
        {...props}
        ref={ref}
      >
        {hasLabel && (
          <FormLabel
            description={description}
            hasError={hasError}
            htmlFor={id}
            required={required}
            disabled={disabled}
          >
            {label}
          </FormLabel>
        )}
        {options.map((option, index) => (
          <div
            key={`${prefix}-radio-${index}-${option.value}`}
            className="flex items-center gap-2"
            role="presentation"
          >
            <Radio
              id={`${prefix}-radio-${index}-${option.value}`}
              label={option.label}
              value={option.value}
              description={option.description}
              disabled={disabled || option.disabled}
              hasError={hasError}
            />
          </div>
        ))}
        {warnings.length > 0 && (
          <FormMessageList messages={warnings} type="warning" />
        )}
        {hasError && <FormMessageList messages={errors} type="error" />}
      </RadioGroup>
    );
  },
);

export const RadioGroupField =
  withFieldValidation<RadioGroupFieldProps>(RadioGroupFieldRaw);

RadioGroupField.displayName = "RadioGroupField";
