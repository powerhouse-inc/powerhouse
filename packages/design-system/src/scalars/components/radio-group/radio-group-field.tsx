import React, { useId } from "react";
import { FormLabel } from "@/scalars/components/form-label";
import { FormMessageList } from "@/scalars/components/form-message";
import { Radio } from "./radio";
import { RadioGroup } from "./radio-group";

export interface RadioOption {
  label: string;
  value: string;
  description?: string;
  disabled?: boolean;
}

export interface RadioGroupFieldProps {
  className?: string;
  defaultValue?: string;
  description?: string;
  errors?: string[];
  id?: string;
  label?: string;
  name?: string;
  onChange?: (value: string) => void;
  radioOptions: RadioOption[];
  required?: boolean;
  value?: string;
}

export const RadioGroupField: React.FC<RadioGroupFieldProps> = ({
  className,
  defaultValue,
  description,
  errors = [],
  id,
  label,
  name,
  onChange,
  radioOptions = [],
  required = false,
  value,
}) => {
  const prefix = useId();
  const hasLabel = label !== undefined;
  const hasError = errors.length > 0;

  return (
    <RadioGroup
      aria-invalid={hasError}
      aria-required={required}
      aria-label={!hasLabel ? "Radio group" : undefined}
      className={className}
      defaultValue={defaultValue}
      id={id}
      name={name}
      onValueChange={onChange}
      value={value}
    >
      {hasLabel && (
        <FormLabel
          description={description}
          hasError={hasError}
          htmlFor={id}
          required={required}
        >
          {label}
        </FormLabel>
      )}
      {radioOptions.map((option, index) => (
        <div
          key={`${prefix}-radio-${index}-${option.value}`}
          className="flex items-center gap-2.5"
          role="presentation"
        >
          <Radio
            id={`${prefix}-radio-${index}-${option.value}`}
            label={option.label}
            value={option.value}
            description={option.description}
            disabled={option.disabled}
            hasError={hasError}
          />
        </div>
      ))}
      {hasError && <FormMessageList messages={errors} type="error" />}
    </RadioGroup>
  );
};
