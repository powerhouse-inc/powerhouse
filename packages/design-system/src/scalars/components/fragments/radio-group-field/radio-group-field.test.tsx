import { describe, expect, it, vi } from "vitest";
import { RadioGroupField } from "./radio-group-field";
import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";

describe("RadioGroupField Component", () => {
  it("should match snapshot", () => {
    const { asFragment } = render(
      <RadioGroupField radioOptions={[{ label: "Option 1", value: "1" }]} />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render radio options with label", () => {
    render(
      <RadioGroupField
        label="Group Label"
        radioOptions={[
          { label: "Option 1", value: "1" },
          { label: "Option 2", value: "2" },
        ]}
      />,
    );
    expect(screen.getByText("Group Label")).toBeInTheDocument();
    expect(screen.getByText("Option 1")).toBeInTheDocument();
    expect(screen.getByText("Option 2")).toBeInTheDocument();
    expect(screen.getAllByRole("radio")).toHaveLength(2);
  });

  it("should call onChange when a Radio label is clicked", async () => {
    const onChangeMock = vi.fn();
    const user = userEvent.setup();
    render(
      <RadioGroupField
        onChange={onChangeMock}
        radioOptions={[
          { label: "Option 1", value: "1" },
          { label: "Option 2", value: "2" },
        ]}
      />,
    );
    await user.click(screen.getByLabelText("Option 1"));
    expect(onChangeMock).toHaveBeenCalledTimes(1);
    expect(onChangeMock).toHaveBeenCalledWith("1");
  });

  it("should change the value when a different radio is selected", async () => {
    const user = userEvent.setup();
    render(
      <RadioGroupField
        defaultValue="1"
        radioOptions={[
          { label: "Option 1", value: "1" },
          { label: "Option 2", value: "2" },
        ]}
      />,
    );
    const radio1 = screen.getByLabelText("Option 1");
    const radio2 = screen.getByLabelText("Option 2");
    expect(radio1).toBeChecked();
    expect(radio2).not.toBeChecked();
    await user.click(radio2);
    expect(radio1).not.toBeChecked();
    expect(radio2).toBeChecked();
  });

  it("should display description when provided", () => {
    render(
      <RadioGroupField
        label="Group Label"
        description="Group Description"
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    // Check that the info icon fallback is rendered
    const iconFallback = screen.getByTestId("icon-fallback");
    expect(iconFallback).toBeInTheDocument();
    expect(iconFallback).toHaveStyle({ width: "16px", height: "16px" });
  });

  it("should show a warning when provided", () => {
    render(
      <RadioGroupField
        label="Group Label"
        warnings={["Warning message"]}
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    expect(screen.getByText("Warning message")).toBeInTheDocument();
  });

  it("should show an error when provided", () => {
    render(
      <RadioGroupField
        label="Group Label"
        errors={["Error message"]}
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    const radioGroup = screen.getByRole("radiogroup");
    expect(radioGroup).toHaveAttribute("aria-invalid", "true");
    expect(screen.getByText("Error message")).toBeInTheDocument();
  });

  it("should have aria-label when no label is provided", () => {
    render(
      <RadioGroupField radioOptions={[{ label: "Option 1", value: "1" }]} />,
    );
    const radioGroup = screen.getByRole("radiogroup");
    expect(radioGroup).toHaveAttribute("aria-label", "Radio group");
  });

  it("should generate unique ids when not provided", () => {
    render(
      <RadioGroupField radioOptions={[{ label: "Option 1", value: "1" }]} />,
    );
    const radioGroup = screen.getByRole("radiogroup");
    expect(radioGroup.id).toMatch(/^.*-radio-group$/);
  });

  it("should use provided id when specified", () => {
    render(
      <RadioGroupField
        id="custom-id"
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    const radioGroup = screen.getByRole("radiogroup");
    expect(radioGroup).toHaveAttribute("id", "custom-id");
  });

  it("should handle required state", () => {
    render(
      <RadioGroupField
        label="Group Label"
        required
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    const radioGroup = screen.getByRole("radiogroup");
    expect(radioGroup).toHaveAttribute("aria-required", "true");
  });

  it("should handle defaultValue", () => {
    render(
      <RadioGroupField
        defaultValue="2"
        radioOptions={[
          { label: "Option 1", value: "1" },
          { label: "Option 2", value: "2" },
        ]}
      />,
    );
    const selectedRadio = screen.getByLabelText("Option 2");
    expect(selectedRadio).toBeChecked();
  });

  it("should handle controlled value", () => {
    render(
      <RadioGroupField
        value="1"
        radioOptions={[
          { label: "Option 1", value: "1" },
          { label: "Option 2", value: "2" },
        ]}
      />,
    );
    const selectedRadio = screen.getByLabelText("Option 1");
    expect(selectedRadio).toBeChecked();
  });

  it("should apply custom className", () => {
    render(
      <RadioGroupField
        // Custom className for testing purposes
        // eslint-disable-next-line tailwindcss/no-custom-classname
        className="custom-class"
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    const radioGroup = screen.getByRole("radiogroup");
    expect(radioGroup).toHaveClass("custom-class");
  });

  it("should handle multiple warnings", () => {
    render(
      <RadioGroupField
        label="Group Label"
        warnings={["Warning 1", "Warning 2"]}
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    expect(screen.getByText("Warning 1")).toBeInTheDocument();
    expect(screen.getByText("Warning 2")).toBeInTheDocument();
  });

  it("should handle multiple errors", () => {
    render(
      <RadioGroupField
        label="Group Label"
        errors={["Error 1", "Error 2"]}
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    expect(screen.getByText("Error 1")).toBeInTheDocument();
    expect(screen.getByText("Error 2")).toBeInTheDocument();
  });

  it("should show both warnings and errors when provided", () => {
    render(
      <RadioGroupField
        label="Group Label"
        warnings={["Warning message"]}
        errors={["Error message"]}
        radioOptions={[{ label: "Option 1", value: "1" }]}
      />,
    );
    expect(screen.getByText("Warning message")).toBeInTheDocument();
    expect(screen.getByText("Error message")).toBeInTheDocument();
  });
});
