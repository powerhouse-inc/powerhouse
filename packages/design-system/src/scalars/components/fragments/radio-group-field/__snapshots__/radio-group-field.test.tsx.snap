// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`RadioGroupField Component > should match snapshot 1`] = `
<DocumentFragment>
  <form
    id=":r0:"
    novalidate=""
  >
    <div
      aria-invalid="false"
      aria-label="Radio group"
      class="flex flex-col gap-2"
      dir="ltr"
      id=":r1:-radio-group"
      role="radiogroup"
      style="outline-color: none; outline-style: none; outline-width: initial;"
      tabindex="0"
    >
      <div
        class="flex items-center gap-2"
        role="presentation"
      >
        <button
          aria-checked="false"
          aria-disabled="false"
          aria-invalid="false"
          class="group peer aspect-square size-4 rounded-full border border-gray-800 dark:border-gray-400 hover:border-gray-900 dark:hover:border-gray-50"
          data-radix-collection-item=""
          data-state="unchecked"
          id=":r1:-radio-0-1"
          role="radio"
          tabindex="-1"
          type="button"
          value="1"
        />
        <input
          aria-hidden="true"
          name="radio-group"
          style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
          tabindex="-1"
          type="radio"
          value="1"
        />
        <label
          class="inline-flex items-center text-sm font-medium leading-[22px] text-gray-900 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-slate-50 cursor-pointer peer-hover:text-gray-900 dark:peer-hover:text-gray-50"
          for=":r1:-radio-0-1"
        >
          Option 1
        </label>
      </div>
      <div
        class="flex items-center gap-2"
        role="presentation"
      >
        <button
          aria-checked="false"
          aria-disabled="false"
          aria-invalid="false"
          class="group peer aspect-square size-4 rounded-full border border-gray-800 dark:border-gray-400 hover:border-gray-900 dark:hover:border-gray-50"
          data-radix-collection-item=""
          data-state="unchecked"
          id=":r1:-radio-1-2"
          role="radio"
          tabindex="-1"
          type="button"
          value="2"
        />
        <input
          aria-hidden="true"
          name="radio-group"
          style="transform: translateX(-100%); position: absolute; pointer-events: none; opacity: 0; margin: 0px; width: 0px; height: 0px;"
          tabindex="-1"
          type="radio"
          value="2"
        />
        <label
          class="inline-flex items-center text-sm font-medium leading-[22px] text-gray-900 dark:text-gray-400 group-hover:text-gray-900 dark:group-hover:text-slate-50 cursor-pointer peer-hover:text-gray-900 dark:peer-hover:text-gray-50"
          for=":r1:-radio-1-2"
        >
          Option 2
        </label>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
